### PR TITLE
Implement the forwarding logger

### PR DESCRIPTION
**Pull request by swift-kim**
- A logging thread is started on app startup which constantly forwards log messages to a connected host. The thread listens for an incoming TCP connection on a port given as an engine argument.
- Checks the presence of the engine argument `verbose-logging` at the very beginning of `FlutterDesktopRunEngine`.
- Misc changes:
  - Rename `switches` in `FlutterProjectBundle` with `engine_arguments`.
  - Add methods `HasArgument` and `GetArgumentValue` to `FlutterProjectBundle`.
  - Move static functions in `flutter_tizen.cc` into an anonymous namespace.
  - Inline `GetLoggingLevel` and `SetLoggingLevel` and minor cleanups.

Contributes to https://github.com/flutter-tizen/flutter-tizen/issues/260.

You can use https://github.com/flutter-tizen/flutter-tizen/pull/271 to test this change.

### CODE REVIEW

File: shell/platform/tizen/flutter_tizen.cc [hunk 1]
```diff
@@ -13,26 +13,47 @@
 #include "flutter/shell/platform/tizen/logger.h"
 #include "flutter/shell/platform/tizen/public/flutter_platform_view.h"
 
+namespace {
+
 // Returns the engine corresponding to the given opaque API handle.
-static flutter::FlutterTizenEngine* EngineFromHandle(
```
Comment (bwikbs) [inline shell/platform/tizen/flutter_tizen.cc:17]: I still didn't understand why this has the same as before.(Even though I read the article you gave me)

**Comment by bwikbs** [comment]
I want to ask one stupid question.(you've already checked it..)
Is this trying to solve some security issue? Can we use the port there?

File: shell/platform/tizen/flutter_tizen.cc (hunk 1 above)
Comment (swift-kim) [reply shell/platform/tizen/flutter_tizen.cc:17]: Article? If I understand you correctly, this is just a coding style difference. In short, `static` is C style and the anonymous namespace is C++ style. I prefer using an anonymous namespace if there are more than one or two static variables/functions in a file.

**Comment by swift-kim** [comment]
> Is this trying to solve some security issue?

@bwikbs Not really. Apparently some TV devices do not support dlog at all (https://github.com/flutter-tizen/flutter-tizen/issues/260) so I'm just trying to deprecate any use of dlog in the flutter-tizen tool. You can still use dlog for a debugging purpose, by manually typing `sdb dlog ConsoleMessage` in the console.

File: shell/platform/tizen/flutter_project_bundle.cc [hunk 2]
```diff
@@ -67,16 +67,34 @@ FlutterProjectBundle::FlutterProjectBundle(
...
+                                            std::string* value) {
+  auto iter =
+      std::find(engine_arguments_.begin(), engine_arguments_.end(), name);
+  if (iter == engine_arguments_.end()) {
+    return false;
+  }
+  auto next = std::next(iter);
+  if (next == engine_arguments_.end()) {
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply shell/platform/tizen/flutter_project_bundle.cc:93]: `end()` does not mean the last element but indicates "not found".

File: shell/platform/tizen/logger.cc [hunk 3]
```diff
@@ -7,88 +7,191 @@
...
+namespace {
+
+#ifndef __X64_SHELL__
+constexpr char kLogTag[] = "ConsoleMessage";
+
+std::string GetLevelName(int level) {
+  switch (level) {
+    case flutter::kLogLevelDebug:
+      return "D";
+    default:
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply shell/platform/tizen/logger.cc:27]: I wanted the levels to be ordered like D-I-W-E-F.

File: shell/platform/tizen/flutter_project_bundle.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (HakkyuKim) [reply shell/platform/tizen/flutter_project_bundle.cc:93]: The `iter` moves to the next element with `std::next(iter)` after finding the element with `name`. Returning `false` when the next element in "not found" would mean that the last element found is not a viable element.

[the remark this answers is not in the document]
Comment (HakkyuKim) [reply shell/platform/tizen/flutter_project_bundle.cc:93]: The `iter` moves to the next element with `std::next(iter)` after finding the element with `name`. Returning `false` when the next element in "not found" would mean that the last element found is not a viable element.

[the remark this answers is not in the document]
Comment (swift-kim) [reply shell/platform/tizen/flutter_project_bundle.cc:93]: If the arguments are provided as
```
--verbose-logging --tizen-logging-port 12345
```
then the call to this function `GetArgumentValue("--tizen-logging-port")` will return `"12345"`. Did you understand correctly? Note that this function doesn't currently support the equal sign form like:
```
--tizen-logging-port=12345
```

[the remark this answers is not in the document]
Comment (HakkyuKim) [reply shell/platform/tizen/flutter_project_bundle.cc:93]: Ah got it, thanks.

File: shell/platform/tizen/logger.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (HakkyuKim) [reply shell/platform/tizen/logger.cc:27]: Okay.

**Review by HakkyuKim — APPROVED** [review]
LGTM

**Review by bwikbs — APPROVED** [review]
Thanks :+1:

File: shell/platform/tizen/logger.h [hunk 4]
```diff
@@ -21,27 +21,32 @@ constexpr int kLogLevelFatal = 4;
...
+  static void Print(int level, const std::string& message);
 
  private:
   explicit Logger();
 
   static void* Redirect(void* arg);
+  static void* Forward(void* arg);
 
-  static inline bool started_ = false;
+  static inline bool is_running_ = false;
```
Comment (bbrto21) [inline shell/platform/tizen/logger.h:40]: how about using atomic?

**Review by bbrto21 — APPROVED** [review]
It works well with https://github.com/flutter-tizen/flutter-tizen/pull/271

File: shell/platform/tizen/logger.h (hunk 4 above)
Comment (swift-kim) [reply shell/platform/tizen/logger.h:40]: I think it makes very much sense to use `std::atomic<bool>` but you already merged the commit =)

**Pull request merged**